use-cases/using-SRIOV-and-kata.md

```diff
@@ -0,0 +1,313 @@
+   See [PCIE-ACS-override-option](https://lkml.org/lkml/2013/5/30/513) for
+   detailed information about this option.
+
+## Set up the SR-IOV Device
+
+All the steps in prior sections need to be performed just once to prepare the 
+SR-IOV host systems. The following will be needed per system boot in order to
+facilitate setting up a physical device's virtual functions.
+
+The following procedure sets up your SR-IOV device and needs to be done per 
+system boot. Set up includes loading a device driver, finding out how many 
+virtual functions (VF) you can create, and creating those virtual functions.
+Once you create VFs you cannot increase or decrease the number of VFs without 
+first setting the number back to zero. Based on this, it is expected that you
+set the number of VFs for a physical device just once.
+
+1. Add `vfio-pci` device driver
+
+   `vfio-pci` is a driver used to reserve a VF PCI device.  Add it:
+   ```
+   $ sudo modprobe vfio-pci
+   ```
+
+2. Find the NICs of interest
+
+   Find PCI details for the NICs in question:
+   ```
+   $ lspci | grep Ethernet
+   00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+   01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   ```
+   In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+   that we will use. You can use `lshw` command to get further details on the 
+   controller and verify it supports SR-IOV.
+
+3. Check how many VFs you can create:
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+   The previous commands show how many VFs you can create. The `sriov_totalvfs` 
+   file under sysfs for a PCI device specifies the total number of VFs that you
+   can create.
+
+4. Create the VFs:
+
+   ```
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+   ```
+   
+   Create virtual functions by editing `sriov_numvfs`. In our example, we create
+   commands create virtual functions by editing `sriov_numvfs`. This example 
+   creates one VF per physical device. Note, creating one VF eliminates the 
+   usefulness of SR-IOV, and is done for simplicity in this example. 
+
+ 5. Verify the VFs were added to the host:
+
+   ```
+   # lspci | grep Ethernet | grep Virtual
+   02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   ```
+
+6. Assign a MAC address to each VF:
+   
+   ```
+   $ ip link set <pf> vf <vfidx> mac <fake MAC address>
+   ```
+
+   Depending on the NIC being used, you  might need to explicitly set the MAC 
```

Comment:
Remove extra space after "you": Depending on the NIC being used, you might need to explicitly set the MAC